beetsplug/convert.py

```diff
@@ -382,12 +393,24 @@ def convert_item(
                 with _fs_lock:
                     util.mkdirall(dest)
 
-            if os.path.exists(util.syspath(dest)):
-                self._log.info(
-                    "Skipping {0} (target file exists)",
-                    util.displayable_path(item.path),
-                )
-                continue
+            if os.path.exists(dest):
+                # Delete existing destination files when original files have
+                # been modified since the last convert run.
+                if refresh and os.path.getmtime(original) > os.path.getmtime(
+                    dest
+                ):
+                    self._log.info(
+                        "Removing {0} (original file modified)",
+                        util.displayable_path(dest),
+                    )
+                    if not pretend:
```

Comment:
Thanks for tidying this up. It's much more clear now, where the logic is clear.

The only question I have left: what happens when `pretend=True`?

As far as I can see, the file will not be removed and the logic will continue on line 415. Before, the logic would never reach line 415 if the file existed. 

Is this intentional?